dashboard/src/components/AppUpgrade/AppUpgrade.tsx

```diff
@@ -41,13 +47,17 @@ interface IAppUpgradeState {
 
 class AppUpgrade extends React.Component<IAppUpgradeProps, IAppUpgradeState> {
   public componentDidMount() {
-    const { releaseName, getApp, namespace, fetchRepositories } = this.props;
-    getApp(releaseName, namespace);
+    const { releaseName, getAppWithUpdateInfo, namespace, fetchRepositories } = this.props;
+    getAppWithUpdateInfo(releaseName, namespace);
     fetchRepositories();
   }
 
   public render() {
-    const { app, repos, error, namespace, releaseName, repoError } = this.props;
+    const { app, repos, error, namespace, releaseName, repoError, isFetching } = this.props;
+    let { repo } = this.props;
+    if (isFetching) {
```

Comment:
it's necessary to check if we are still fetching things to avoid the `SelectRepoForm` to be momentarily shown.